src/update.c

```diff
@@ -367,6 +367,55 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Get a list of all the iterative manifests from the server_manifest that match
+	* the "from" version, also get a list of all the full manifests in the MoM */
+	struct list *iterative_manifests = NULL;
+	struct list *full_manifests = NULL;
+	struct list *list1;
+	struct file *file1;
+
+	list1 = server_manifest->manifests;
+	while (list1) {
+		file1 = list1->data;
+		list1 = list1->next;
+		if (file1->is_iterative == 1) {
+			/* Discard iterative manifests that are not from the current version */
+			if (file1->from_version == current_version) {
+				iterative_manifests = list_prepend_data(iterative_manifests, file1);
+			}
+		} else {
+			full_manifests = list_prepend_data(full_manifests, file1);
+		}
+	}
+
+	/* Add the full manifest of those bundles that do not have a valid iterative manifest
+	* into the list of iterative_manifests */
+	struct list *list2;
+	struct file *file2;
+	bool add_manifest;
+
+	list1 = full_manifests;
+	while (list1) {
```

Comment:
If we don't already have a helper for finding a manifest in a list of manifests you should add one instead of embedding the logic here.

Reply:
That is a good idea, I will add a search_bundle_in_manifests function to manifest.c (I couldn't find one)